src/Extension/TextExtension.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Faker\Core\Extension;
+
+/**
+ * @experimental This interface is experimental and does not fall under our BC promise
+ */
+interface TextExtension extends Extension
+{
+    /**
+     * Generate a text string by the Markov chain algorithm.
+     *
+     * Depending on the $maxNbChars, returns a random valid looking text. The algorithm
+     * generates a weighted table with the specified number of words as the index and the
+     * possible following words as the value.
+     *
+     * @example 'Alice, swallowing down her flamingo, and began by taking the little golden key'
+     *
+     * @param int $minNbChars Minimum number of characters the text should contain (maximum: 8)
+     * @param int $maxNbChars Maximum number of characters the text should contain (minimum: 10)
+     * @param int $indexSize  Determines how many words are considered for the generation of the next word.
+     *                        The minimum is 1, and it produces a higher level of randomness, although the
+     *                        generated text usually doesn't make sense. Higher index sizes (up to 5)
+     *                        produce more correct text, at the price of less randomness.
+     *
+     * @return string
+     */
+    public function realTextBetween(int $minNbChars = 160, int $maxNbChars = 200, int $indexSize = 2): string;
```

Comment:
I would prefer to use a more clean parameter names like `min` and `max`. 
The docblock does plenty explaining of their function I think